### PR TITLE
fix(append): support Message.experimental_attachments

### LINKED ISSUE #4908: `experimental_attachments` doesn't work in first argument to `append`
### Description

```
const {
    messages,
    isLoading: isGenerating,
    append: generateDescription,
    error: generationError,
    setMessages
  } = useChat({
    sendExtraMessageFields: true,
    api: '/api/generate-listing-parts',
    id: section.id,
    body: {
      projectId: project_id,
      sectionId: section.id
    },
    onFinish: (message, {usage, finishReason}) => {
      console.log('Finished streaming message:', message)
      console.log('Token usage:', usage)
      console.log('Finish reason:', finishReason)
    },
    onError: error => {
      console.error('An error occurred:', error)
    },
    onResponse: response => {
      console.log('Received HTTP response from server:', response)
    }
  })
```

then later down

```
await generateDescription({
          role: 'user',
          content: '',
          experimental_attachments: [...photos, ...uploadedPhotos].map(
            image => ({
              url: image.url,
              type: image.type
            })
          )
        })
```

for some reason the `experimental_attachments` are not sent 

### Code example

_No response_

### AI provider

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by samdenty**
## Background

We didn't support `Message.experimental_attachments` which is confusing since you have to pass it in a second argument

fixes #4908

## Summary

Added support for `Message.experimental_attachments`

## Verification

E2E test

## Tasks

- [x] Tests have been added / updated (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] A _patch_ changeset for relevant packages has been added (for bug fixes / features - run `pnpm changeset` in the project root)
- [x] Formatting issues have been fixed (run `pnpm prettier-fix` in the project root)

### CODE REVIEW

File: packages/vue/src/use-chat.ts [hunk 1]
```diff
@@ -317,7 +317,7 @@ export function useChat(
 
   const append: UseChatHelpers['append'] = async (message, options) => {
     const attachmentsForRequest = await prepareAttachmentsForRequest(
-      options?.experimental_attachments,
```
Comment (samdenty) [inline packages/vue/src/use-chat.ts:320]: i could also add a vue test, but may have to duplicate the file

**Comment by lgrammel** [comment]
What do you mean by verification: E2E test? did you e.g. run a manual example under `examples/`?

**Review by lgrammel — APPROVED** [review]

**Comment by samdenty** [comment]
> What do you mean by verification: E2E test

I added a test and it passed with the new behaviour that was failing before
